src/I18n/Parser/PoFileParser.php

```diff
@@ -142,11 +142,10 @@ protected function _addMessage(array &$messages, array $item)
 
         $translation = stripcslashes($translation);
 
-        if ($context && (!isset($messages[$singular]) || is_array($messages[$singular]))) {
+        if ($context !== null) {
             $messages[$singular]['_context'][$context] = $translation;
-        }
-        if ($context === null) {
-            $messages[$singular] = $translation;
+        } else {
+            $messages[$singular]['_context'][''] = $translation;
```

Comment:
I wonder what the memory impact of this change will be like on larger applications.